src/DataSource.ts

```diff
@@ -97,31 +95,65 @@ export class DataSource extends DataSourceApi<AkenzaQuery, AkenzaDataSourceConfi
             }
         }
 
-        return { data: panelData };
+        return {data: panelData};
     }
 
     async getDevices(searchString?: string): Promise<Device[]> {
-        return this.getOrganization().then((organization) => {
-            return new Promise((resolve, reject) => {
-                const params = {
-                    organizationId: organization.id,
-                    type: 'DEVICE',
-                    search: searchString,
-                };
-
-                this.executeRequest<DeviceList>(
-                    '/v3/assets',
-                    'GET',
-                    params)
-                    .subscribe({
-                        next(response: FetchResponse<DeviceList>) {
-                            resolve(response.data.content)
-                        },
-                        error(error: FetchError) {
-                            reject(DataSource.generateErrorMessage(error))
-                        }
-                    });
-            });
+        const organization = await this.getOrganization();
+        const access = await this.checkWorkspaceAccess(organization.id);
+
+        const params: AccessCheckParams = {
```

Comment:
the type of this should likely be AssetFilterParams

Reply:
ah yes, the old name confusion ^^